crates/ty_python_semantic/src/semantic_model.rs

```diff
@@ -215,9 +215,10 @@ impl<'db> Completion<'db> {
                 | Type::Callable(_) => CompletionKind::Function,
                 Type::BoundMethod(_) | Type::MethodWrapper(_) => CompletionKind::Method,
                 Type::ModuleLiteral(_) => CompletionKind::Module,
-                Type::ClassLiteral(_) | Type::GenericAlias(_) | Type::SubclassOf(_) => {
-                    CompletionKind::Class
-                }
+                Type::ClassLiteral(_)
+                | Type::GenericAlias(_)
+                | Type::SubclassOf(_)
+                | Type::TypedDict(_) => CompletionKind::Class,
```

Comment:
`TypedDict` feels a bit different to the other ones in this branch -- the others are all class objects (or similar) whereas `Type::TypedDict` is more similar to `Type::NominalInstance` or `Type::ProtocolInstance` IIUC, in that it inhabits the type defined by a class rather than actually _being_ the class (or a subclass of the class)

I wonder if calling the variant `TypedDictInhabitant` might make this clearer?